nupic/research/frameworks/vernon/handlers.py

```diff
@@ -1059,6 +1058,368 @@ def get_execution_order(cls):
+        which are defined in SupervisedExperiment, but some of which are specific to
+        MetaContinualLearningExperiment
+
+            - experiment_class: Class used to run experiments, specify
+            `MetaContinualLearningExperiment` for meta-continual learning
+            - adaptation_lr: Learning rate used to update the fast parameters during
+            the inner loop of the meta-training phase
+            - tasks_per_epoch: Number of different classes used for training during the
+            execution of the inner loop
+            - slow_batch_size: Number of examples in a single batch used to update the
+            slow parameters during the meta-training testing phase, where the examples
+            are sampled from tasks_per_epoch difference tasks
+            - replay_batch_size: Number of examples in a single batch sampled from all
+            data, also used to train the slow parameters (the replay batch is used to
+            sample examples to update the slow parameters during meta-training testing
+            to prevent the learner from forgetting other tasks)
+        """
+        if "num_classes" not in config["model_args"]:
+            # manually set `num_classes` in `model_args`
+            num_classes = config["num_classes"]
+            config["model_args"]["num_classes"] = num_classes
+
+        super().setup_experiment(config)
+        self.train_slow_loader = self.train_loader
+        self.train_fast_loader, self.val_fast_loader, self.train_replay_loader = \
+            self.create_fast_slow_loaders(config)
+
+        self.epochs_to_validate = []
+        self.tasks_per_epoch = config.get("tasks_per_epoch", 1)
+        self.num_classes = min(
+            config.get("num_classes", 50),
+            self.train_fast_loader.sampler.num_classes
+        )
+
+        self.adaptation_lr = config.get("adaptation_lr", 0.03)
+
+        self.batch_size = config.get("batch_size", 5)
+        self.val_batch_size = config.get("val_batch_size", 15)
+        self.slow_batch_size = config.get("slow_batch_size", 64)
+        self.replay_batch_size = config.get("replay_batch_size", 64)
+
+    @classmethod
+    def create_loaders(cls, config):
+        """Create train and val dataloaders."""
+
+        dataset_class = config.get("dataset_class", None)
+        if dataset_class is None:
+            raise ValueError("Must specify 'dataset_class' in config.")
+
+        # Create datasets -> same, only initialize two of them
+        dataset_args = config.get("dataset_args", {})
+        dataset_args.update(train=True)
+        main_set = dataset_class(**dataset_args)
+
+        # All loaders share tasks and dataset, but different indices and batch sizes
+        train_slow_loader = cls.create_slow_train_dataloader(main_set, config)
+
+        # For pre/post epoch and batch processing slow loader is equiv to train loader
+
+        # note: modify return values? right now, it's returning the torch dataloader
```

Comment:
I am not sure what the comment is about, maybe the changes Karan introduced in one of the latest commits. But the way I had setup I don't see that issue there - both functions have the same signature, they don't violate the substitution principle.

Please review the code and see if you still have the same concern.

Reply:
This comment was mainly about the `create_loaders` before you last commit (330a1af). I think that commit solves a big part of the problem.

I'm also curious whether the fact we don't use `evaluate` points to another problem. That is, maybe since we don't use it, it means it's not technically a supervised experiment. I'm not sure about this yet. Regardless, it's not an issue for this PR and can be addressed later on after the introduction of a `BaseExperiment`.